smpmgr/image_management.py

```diff
@@ -71,7 +71,39 @@ def state_write(
     async def f() -> None:
         await connect_with_spinner(smpclient, options.timeout)
 
-        r = await smp_request(smpclient, options, ImageStatesWrite(hash=hash_bytes, confirm=confirm), "Waiting for image state write...")  # type: ignore # noqa
+        r = await smp_request(
+            smpclient,
+            options,
+            ImageStatesWrite(hash=hash_bytes, confirm=confirm),
+            "Waiting for image state write...",
+        )
+
+        if error(r):
+            print(r)
+        elif success(r):
+            pass
+        else:
+            raise Exception("Unreachable")
+
+    asyncio.run(f())
+
+
+@app.command()
+def erase(
+    ctx: typer.Context,
+    slot: Annotated[int, typer.Option(help="The image slot to erase")] = 1,
```

Comment:
I missed this. This is such a destructive command that I think it's better not to have any defaults. The user really needs to look at the output of the read to get the slot arg.

Come to think of it, this is another inconstancy in the spec. Write requires the hash, making it unambiguous, yet erase uses the slot.

Let's
- remove the default
- update the argument help to explain that slot is the slot from image state read, "0-indexed".

Reply:
dun